### PR TITLE
ADEN-4124 mercury

### PR DESCRIPTION
**Pull request by v3rth**
This task is on the top of [ADEN-4124](https://github.com/Wikia/app/tree/ADEN-4124)
• expand video on click
• hide video after playing

### CODE REVIEW

File: extensions/wikia/AdEngine/js/template/bfaaDesktop.js [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+	'ext.wikia.adEngine.adHelper',
+	'ext.wikia.adEngine.context.uapContext',
+	'ext.wikia.adEngine.provider.btfBlocker',
+	'ext.wikia.adEngine.slotTweaker',
+	'ext.wikia.adEngine.video.uapVideoAd',
+	'wikia.document',
+	'wikia.log',
+	'wikia.window',
+	require.optional('ext.wikia.aRecoveryEngine.recovery.tweaker')
+], function (
```
Comment (WikiaTech) [inline extensions/wikia/AdEngine/js/template/bfaaDesktop.js]: MAJOR Function has 9 parameters which is greater than 7 authorized. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AExcessiveParameterList)

File: extensions/wikia/AdEngine/js/template/bfaaMobile.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+	'ext.wikia.adEngine.adHelper',
+	'ext.wikia.adEngine.context.uapContext',
+	'ext.wikia.adEngine.provider.btfBlocker',
+	'ext.wikia.adEngine.slotTweaker',
+	'ext.wikia.adEngine.video.uapVideoAd',
+	'wikia.document',
+	'wikia.log',
+	'wikia.window',
+	require.optional('ext.wikia.adEngine.mobile.mercuryListener')
+], function (
```
Comment (WikiaTech) [inline extensions/wikia/AdEngine/js/template/bfaaMobile.js]: MAJOR Function has 9 parameters which is greater than 7 authorized. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AExcessiveParameterList)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues
* MAJOR 2 major

Watch the comments in this conversation to review them.

File: extensions/wikia/AdEngine/js/template/bfaaDesktop.js [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+		}
+
+		if (recoveryTweaker && recoveryTweaker.isTweakable()) {
+			slotTweaker.removeDefaultHeight(params.slotName);
+			recoveryTweaker.tweakSlot(params.slotName, iframe);
+		}
+
+		if (params.videoUrl && params.videoTriggerElement) {
+			slotTweaker.onReady(params.slotName, function(iframe) {
+				var imageContainer = iframe.parentNode.parentNode.parentNode,
```
Comment (fraszczakszymon) [inline extensions/wikia/AdEngine/js/template/bfaaDesktop.js]: Can we implement it in different way?

File: extensions/wikia/AdEngine/js/template/bfaaMobile.js [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+				page.classList.remove('bfaa-template');
+				page.style.paddingTop = '';
+				adsModule.setSiteHeadOffset(0);
+				win.removeEventListener('resize', onResize);
+			});
+		}
+
+		if (params.videoUrl && params.videoTriggerElement) {
+			slotTweaker.onReady(params.slotName, function(iframe) {
+				var imageContainer = iframe.parentNode.parentNode.parentNode,
```
Comment (fraszczakszymon) [inline extensions/wikia/AdEngine/js/template/bfaaMobile.js]: Can we implement it in different way?

Comment (drets) [reply extensions/wikia/AdEngine/js/template/bfaaMobile.js]: • get top leaderboard by slot name and then get first div child
• walk up from iframe till parenNode equals top leaderboard
What do you prefer, any other ideas?

Comment (fraszczakszymon) [reply extensions/wikia/AdEngine/js/template/bfaaMobile.js]: get top leaderboard by slot name and then get first not-hidden div child

File: extensions/wikia/AdEngine/js/template/bfaaMobile.js [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+			mercuryListener.onPageChange(function () {
+				page.classList.remove('bfaa-template');
+				page.style.paddingTop = '';
+				adsModule.setSiteHeadOffset(0);
+				win.removeEventListener('resize', onResize);
+			});
+		}
+
+		if (params.videoUrl && params.videoTriggerElement) {
+			slotTweaker.onReady(params.slotName, function(iframe) {
```
Comment (WikiaTech) [inline extensions/wikia/AdEngine/js/template/bfaaMobile.js]: MAJOR Remove the unused function parameter "iframe". [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 4 issues
* MAJOR 4 major

Watch the comments in this conversation to review them.

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues
* MAJOR 2 major

Watch the comments in this conversation to review them.

**Comment by bognix** [comment]
lgtm

**Comment by bognix** [comment]
resolve conflicts

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:
1. MAJOR [bfaaDesktop.js#L12](https://github.com/Wikia/app/blob/20722729173c77735e913966a1b44149668f7913/extensions/wikia/AdEngine/js/template/bfaaDesktop.js#L12): Function has 9 parameters which is greater than 7 authorized. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AExcessiveParameterList)
2. MAJOR [bfaaMobile.js#L12](https://github.com/Wikia/app/blob/20722729173c77735e913966a1b44149668f7913/extensions/wikia/AdEngine/js/template/bfaaMobile.js#L12): Function has 9 parameters which is greater than 7 authorized. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AExcessiveParameterList)

**Pull request merged**